tests/test_operation.py

```diff
@@ -2212,6 +2293,68 @@ def test_expansion_broadcasted(self):
         )
         assert np.allclose(expected, res)
 
+        res = qml.operation.expand_matrix(
+            self.base_matrix_3_broadcasted, wires=[2], wire_order=[0, 2]
+        )
+        expected = np.array(
+            [
+                [
+                    [1, 2, 3, 0, 0, 0, 0, 0, 0],
+                    [4, 5, 6, 0, 0, 0, 0, 0, 0],
+                    [7, 8, 9, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 1, 2, 3, 0, 0, 0],
+                    [0, 0, 0, 4, 5, 6, 0, 0, 0],
+                    [0, 0, 0, 7, 8, 9, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 1, 2, 3],
+                    [0, 0, 0, 0, 0, 0, 4, 5, 6],
+                    [0, 0, 0, 0, 0, 0, 7, 8, 9],
+                ],
+                [
+                    [10, 11, 12, 0, 0, 0, 0, 0, 0],
+                    [13, 14, 15, 0, 0, 0, 0, 0, 0],
+                    [16, 17, 18, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 10, 11, 12, 0, 0, 0],
+                    [0, 0, 0, 13, 14, 15, 0, 0, 0],
+                    [0, 0, 0, 16, 17, 18, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 10, 11, 12],
+                    [0, 0, 0, 0, 0, 0, 13, 14, 15],
+                    [0, 0, 0, 0, 0, 0, 16, 17, 18],
+                ],
+            ]
+        )
+        assert np.allclose(expected, res)
+
+        res = qml.operation.expand_matrix(
+            self.base_matrix_3_broadcasted, wires=[2], wire_order=[2, 0]
+        )
+        expected = np.array(
+            [
+                [
+                    [1, 0, 0, 2, 0, 0, 3, 0, 0],
+                    [0, 1, 0, 0, 2, 0, 0, 3, 0],
+                    [0, 0, 1, 0, 0, 2, 0, 0, 3],
+                    [4, 0, 0, 5, 0, 0, 6, 0, 0],
+                    [0, 4, 0, 0, 5, 0, 0, 6, 0],
+                    [0, 0, 4, 0, 0, 5, 0, 0, 6],
+                    [7, 0, 0, 8, 0, 0, 9, 0, 0],
+                    [0, 7, 0, 0, 8, 0, 0, 9, 0],
+                    [0, 0, 7, 0, 0, 8, 0, 0, 9],
+                ],
+                [
+                    [10, 0, 0, 11, 0, 0, 12, 0, 0],
+                    [0, 10, 0, 0, 11, 0, 0, 12, 0],
+                    [0, 0, 10, 0, 0, 11, 0, 0, 12],
+                    [13, 0, 0, 14, 0, 0, 15, 0, 0],
+                    [0, 13, 0, 0, 14, 0, 0, 15, 0],
+                    [0, 0, 13, 0, 0, 14, 0, 0, 15],
+                    [16, 0, 0, 17, 0, 0, 18, 0, 0],
+                    [0, 16, 0, 0, 17, 0, 0, 18, 0],
+                    [0, 0, 16, 0, 0, 17, 0, 0, 18],
+                ],
+            ]
+        )
+        assert np.allclose(expected, res)
+
```

Comment:
I notice there are a number of other tests in this test class, e.g., expansion in operator class; maybe someone on the Xanadu side can weigh in on whether those should have qutrit examples as well? Given that the only change was the base dimensionality and nothing more fundamental in the function, I'm inclined to say they are not needed. They also aren't parametrized tests so adding more cases is non-trivial.

Reply:
Ideally, we'd like to test every component that uses the qutrit pipeline, even if it is shared with the qubit implementation. Something unexpected could start happening and such tests will guide us.

> I notice there are a number of other tests in this test class, e.g., expansion in operator class;

Not quite sure which tests these are in specific, but definitely worth adding qutrit variants if such cases are not otherwise covered.

Reply:
It was tests like `test_expand_two_consecutive_wires`, `test_expand_two_reversed_wires`, etc. Basically all the tests in the class that occur after the interface-specific tests.